src/policy/rbf.cpp

```diff
@@ -60,57 +59,26 @@ std::optional<std::string> GetEntriesForConflicts(const CTransaction& tx,
 {
     AssertLockHeld(pool.cs);
     const uint256 txid = tx.GetHash();
-    uint64_t nConflictingCount = 0;
-    for (const auto& mi : iters_conflicting) {
-        nConflictingCount += mi->GetCountWithDescendants();
-        // Rule #5: don't consider replacing more than MAX_REPLACEMENT_CANDIDATES
-        // entries from the mempool. This potentially overestimates the number of actual
-        // descendants (i.e. if multiple conflicts share a descendant, it will be counted multiple
-        // times), but we just want to be conservative to avoid doing too much work.
-        if (nConflictingCount > MAX_REPLACEMENT_CANDIDATES) {
-            return strprintf("rejecting replacement %s; too many potential replacements (%d > %d)\n",
-                             txid.ToString(),
-                             nConflictingCount,
-                             MAX_REPLACEMENT_CANDIDATES);
-        }
-    }
     // Calculate the set of all transactions that would have to be evicted.
     for (CTxMemPool::txiter it : iters_conflicting) {
-        pool.CalculateDescendants(it, all_conflicts);
-    }
-    return std::nullopt;
-}
-
-std::optional<std::string> HasNoNewUnconfirmed(const CTransaction& tx,
-                                               const CTxMemPool& pool,
-                                               const CTxMemPool::setEntries& iters_conflicting)
-{
-    AssertLockHeld(pool.cs);
-    std::set<uint256> parents_of_conflicts;
-    for (const auto& mi : iters_conflicting) {
-        for (const CTxIn& txin : mi->GetTx().vin) {
-            parents_of_conflicts.insert(txin.prevout.hash);
+        // Exit early if we're going to fail (see below)
+        if (all_conflicts.size() > MAX_REPLACEMENT_CANDIDATES) {
```

Comment:
There is room to relax this rule some, so if this is important we can do so. I think the requirement is a bound on the number of clusters that would have to be re-sorted in order to accept the new transaction.  We can approximate that as the number of clusters that would be non-empty as a result of removing all the conflicting transactions from the mempool, and only process replacements for which that is below some target.  

That would be a more complex logic though, so before implementing it I wanted to have some sense of whether we need to.  Has the historical 100-transaction-conflict limit been problematic for use cases in the past?  Note also that in the new code, we are calculating the number of conflicts exactly (the old code used an approximation, which could be gamed by an adversary).

Reply:
Ah! I wrote a huge response to this, then looked up our previous discussions, and realized I didn't actually read the code: https://github.com/bitcoin/bitcoin/issues/27677#issuecomment-1554708843

IIUC now, this is only counting *direct* conflicts, and not the descendants that are booted.

I think that's fine.

Reply:
Ah yes, in the new code I'm only counting direct conflicts right now, because every descendant of a direct conflict must be in the same cluster as that conflict.  So this is already a relaxation of the existing rule.

Reply:
> I think the requirement is a bound on the number of clusters that would have to be re-sorted in order to accept the new transaction.

As an alternative, we drop the replacement limit to like, 10 or something, and then *only* count the *direct* conflicts, not the direct conflicts and all the descendants?

Reply:
I believe I (finally) actually fixed this behavior to count the number of direct conflicts.

Reply:
Will comment elsewhere as well, but I just updated this limit to be a limit on the number of distinct clusters that are conflicted.